src/styles/blocks/stars.scss

```diff
@@ -5,7 +5,7 @@
 
   margin-right: 16px;
 
-  &--4 :nth-child(-n+4) {
+  & :nth-child(-n+4) {
```

Comment:
In this case your cards have always 4 active stars, but you need has modificator --4 that set active stars

stars-block--2 - 2 active stars
stars-block--4 - 4 active stars
stars-block--5 - 5